src/lib/debug/debug_uart.c

```diff
@@ -285,6 +286,92 @@ static BaseType_t debugSerialCommand(char *writeBuffer,
       // Transmit bytes
       // should have generic tx task for all interfaces?
 
+    } else if (strncmp("rxLevel", parameter, parameterStringLength) == 0) {
+
+      parameter = FreeRTOS_CLIGetParameter(
+                      commandString,
+                      2, // Get the second parameter (serial interface)
+                      &parameterStringLength);
+      SerialHandle_t *handle = serialFindInterface(parameter, parameterStringLength);
+
+      if (handle == NULL) {
+        printf("ERR Invalid interface\n");
+        break;
+      }
+
+      parameter = FreeRTOS_CLIGetParameter(
+                      commandString,
+                      3, // Get the rest of the parameters
+                      &parameterStringLength);
+
+      LL_USART_Disable((USART_TypeDef *)handle->device);
```

Comment:
Intersting thing to note here, the reference manual says the usart/lpuart needs to be disabled (aka the bit this func sets = 0) in order to change the CR2 register (which tells the peripheral the poloarity of the tx/rx pins. That is why I am doing this here. The interesting part is that the reference manual also says the usart/lpuart needs to be disabled when changing the baud register BRR, however, from experience... it dosen't.

Reply:
🤔 🤷‍♂️